src/services/rate-backend.ts

```diff
@@ -12,6 +12,7 @@ import { loadModuleOfType } from '../lib/utils'
 const DEFAULT_BACKEND = 'one-to-one'
 
 export default class RateBackend implements BackendInstance {
+  public submitPacket: any
```

Comment:
Use the callback's fn signature here.

Reply:
Resolved by comment:

> Instead of conditionally attaching this function and then having callers have to test for it, why not add a method to RateBackend that does submitPacket(...) { if (this.backend.submitPacket) { this.backend.submitPacket(...) }}. submitPayemnt could do the same.